cinema/views.py

```diff
@@ -19,27 +20,48 @@
     OrderSerializer,
     OrderListSerializer,
 )
+from user.permissions import IsAdminOrIfAuthenticatedReadOnly
 
 
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
+
 
 class ActorViewSet(viewsets.ModelViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
+
 
 class CinemaHallViewSet(viewsets.ModelViewSet):
     queryset = CinemaHall.objects.all()
     serializer_class = CinemaHallSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
```

Comment:
In `CinemaHallViewSet`, the `get_permissions` method allows 'create' action for authenticated users. Confirm if this is the desired behavior or if it should be restricted to admins.